runtime/domain.c

```diff
@@ -906,13 +906,46 @@ void caml_handle_gc_interrupt() {
   }
 }
 
+CAMLexport void caml_bt_leave_blocking_section_hook(void)
+{
+  dom_internal* self = domain_self;
+
+  if (self->backup_thread_running) {
+    atomic_store_rel(&self->backup_thread_msg, BT_ENTERING_OCAML);
+    caml_plat_lock(&self->interruptor.lock);
+    caml_plat_signal(&self->interruptor.cond);
+    caml_plat_unlock(&self->interruptor.lock);
+  }
+
+  caml_plat_lock(&self->domain_lock);
+
+  return;
+}
+
+CAMLexport void caml_bt_enter_blocking_section_hook(void)
+{
+  dom_internal* self = domain_self;
+
```

Comment:
Can we put back the assert:
`Assert(caml_domain_alone() || self->backup_thread_running);`